source/app/schedulers/instance_scheduler.py

```diff
@@ -137,7 +137,16 @@ def _regions(self):
     @property
     def _sts(self):
         if self._sts_client is None:
-            self._sts_client = get_client_with_standard_retry("sts")
+            session = boto3.Session()
+            sts_regional_endpoint = str.format(
+                "https://sts.{}.amazonaws.com", session.region_name
+            )
+            self._sts_client = session.client(
+                "sts",
+                region_name=session.region_name,
+                endpoint_url=sts_regional_endpoint,
+            )
```

Comment:
If you're not going to use the helper to get a client, you need to be sure to give this the same behavior. Notably, you have not set the user agent extra string, nor have you configured retries.

Reply:
just said the same thing myself XD

if we need to maintain retries for this, I'll probably need a little help understanding the helper function. It's a little unclear exactly how the retries are added. (appears to me like it's set somewhere higher up the call chain)

Reply:
No need to refactor the helper if that's difficult. Just specify retries and user agent extra here too.

Reply:
Took another look at it and refactored part of the helper such that I could extract the default retries configuration for re-use here